tests/Test.ADAL.NET.Unit.net45/UnitTests.cs

```diff
@@ -117,6 +119,30 @@ public void AuthorityTypeDetectionTest()
             Assert.AreEqual(AuthorityType.ADFS, Authenticator.DetectAuthorityType("https://abc.com/adfs/dummy/"));
         }
 
+        [TestMethod]
+        public void AuthorityCustomPortTest()
+        {
+            using (var httpManager = new MockHttpManager())
+            {
+                var CustomPortAuthority = "https://localhost:5215/common/";
+                httpManager.AddInstanceDiscoveryMockHandler();
+                var serviceBundle = ServiceBundle.CreateWithCustomHttpManager(httpManager);
+
+                var context = new AuthenticationContext(
+                              serviceBundle,
+                              CustomPortAuthority,
+                              AuthorityValidationType.False,
+                              new TokenCache());
```

Comment:
You should asssert the authority here as well: 

` Assert.AreEqual(CustomPortAuthority, context.Authority);`